github-actions/unified-status-check/src/pull-request.ts

```diff
@@ -5,59 +5,146 @@ import {CheckConclusionState, PullRequestState, StatusState} from '@octokit/grap
-      if (ignored.some((matcher) => status.name.match(matcher))) {
-        this.ignored.push(status);
-        if (status.name === unifiedStatusCheckName) {
-          this.unifiedCheckStatus = status;
+  /** Normalize the statuses and checks for usage on the pull request, and then populate the statuses values. */
+  private normalizeAndPopulateStatuses(checksAndStatuses: CheckOrStatus[] = []) {
+    checksAndStatuses.forEach((checkOrStatus: CheckOrStatus) => {
+      /** The normalized status. */
+      let status: NormalizedStatus;
+
+      if (checkOrStatus.__typename === 'CheckRun') {
+        status = {
+          state: checkOrStatus.conclusion
+            ? checkConclusionStateToStatusStateMap.get(checkOrStatus.conclusion)!
+            : 'pending',
+          description: checkOrStatus.title || undefined,
+          name: checkOrStatus.name || 'unknown-check-run',
+        };
+
+        // If the check is the previous result for the unified check, store the previous run id.
+        if (checkOrStatus.name === PullRequest.checkName) {
+          this.previousRunId = checkOrStatus.databaseId;
         }
-        continue;
+      } else if (checkOrStatus.__typename === 'StatusContext') {
+        status = {
+          state: checkConclusionStateToStatusStateMap.get(checkOrStatus.state)!,
+          name: checkOrStatus.context,
+          description: checkOrStatus.description || undefined,
+        };
+      } else {
+        // If the result provided is not a StatusContext or CheckRun, no action should be taken as
+        // we only track statuses and checks.
+        return;
       }
+
+      // If a status is ignored, track it as ignored but not anywhere else.
+      if (PullRequest.ignored.some((matcher) => status.name.match(matcher))) {
+        this.statuses.ignored.push(status);
+        return;
+      }
+
       switch (status.state) {
         case 'failure':
-          this.failing.push(status);
+          this.statuses.failure.push(status);
           break;
         case 'success':
-          this.passing.push(status);
+          this.statuses.success.push(status);
           break;
         case 'pending':
-          this.pending.push(status);
+          this.statuses.pending.push(status);
           break;
       }
 
-      this.all.push(status);
+      this.statuses.all.push(status);
+    });
+  }
+
+  /** Set the check run result for the pull request. */
+  async setCheckResult(state: NormalizedState, description?: string) {
+    /** The parameters for the check run update or creation. */
+    const parameters = {
+      ...context.repo,
+      name: PullRequest.checkName,
+      head_sha: this.sha,
+      status: state === 'pending' ? 'in_progress' : 'completed',
+      conclusion: state === 'pending' ? undefined : state,
+      output: {
+        summary: description,
+      },
+    };
+
+    if (this.previousRunId) {
+      await this.github.checks.update({check_run_id: this.previousRunId, ...parameters});
```

Comment:
Ahh, and you changed it from a status to a check 👍 (commit didn't  say that - but nice!)